pkg/nodeconfig/payload/payload.go

```diff
@@ -110,6 +110,39 @@ param(
     }
   }
 
+# This retries getting the HNS-network until it succeeds
+function Retry-GetHnsNetwork {
+     param (
+         [int]$retryCount = 20,
+         [int]$retryDelay = 10
+     )
+ 
+     $attempt = 0
+ 
+     while ($attempt -lt $retryCount) {
+         try {
+             $hns_network = Get-HnsNetwork | Where-Object { $_.Name -eq 'HNS_NETWORK' }
+             
+             # Check if hns_network is null
+             if ($null -eq $hns_network) {
+                 Write-Host "Attempt $($attempt + 1) returned null. Retrying..."
+             } else {
+                 Write-Host "Found HNS_NETWORK on attempt $($attempt + 1)"
+                 return $hns_network 
+             }
+         } catch {
+             Write-Host "Attempt $($attempt + 1) failed: $_"
+             if ($attempt -eq ($retryCount - 1)) {
+                 Write-Host "Max retry attempts reached. Continuing."
+                 exit 1
+             }
```

Comment:
Please move this outside the while
The while will complete at attempt == retry and it should execute this, without needing another comparison.

Reply:
exit 1 doesn't seem correct here.
You should return from within a function and let the caller handle the error, rather than exiting the process from within the function.